src/java/org/apache/cassandra/cql3/Json.java

```diff
@@ -42,7 +42,7 @@
      */
     public static String quoteAsJsonString(String s)
     {
-        return new String(BufferRecyclers.getJsonStringEncoder().quoteAsString(s));
+        return new String(JsonStringEncoder.getInstance().quoteAsString(s));
```

Comment:
Is there a performance impact from these change? (I don't know the difference between `BufferRecyclers` and `JsonStringEncoder`)
My concern is that this is used at a per-column and per-request level, and a performance change could result in bringing a cluster down… (especially if we apply this to release versions)

Reply:
Good question! I was about to say that there is no performance effect, but going back and looking at code it is not that easy to determine actually. Unfortunately `JsonStringEncoder` change was squarely aimed at usage by Jackson-core itself and not for (unplanned) external use. For Jackson's use there should not be measurable difference given where method is called (only from `SerializedString`, used when constructing serializers).


The change did remove use of `ThreadLocal` recycled `TextBuffer` and as a result every encoding does eagerly allocate a `char[120]`; reallocate if needs to expand. This will create bit more garbage than if the underlying working buffer was recycled as in 2.9.

The question then is whether this would have measurable effect, which depends on how often method gets called (like, a hundred times per request or more?) or not.

Given this, I would probably suggest not applying this to 3.x, out of "abundance of caution".

I could otherwise try writing a `jmh` test, but it'd be bit tricky to figure out how to exactly replicate behavior without copying `JsonStringEncoder` implementation (since we'd need different behavior from 2.9 and 2.12).
And without some sense of actual load it is also difficult to gauge impact.

For trunk it might be enough to rely on full stress/load-testing: I doubt there will be severe slowdown (added memory usage is all transient garbage), but in theory there could be increased GC which could lead to minor observable performance regression.

Reply:
Oh. Forgot to add the important piece: change is necessary since 2.10 does not have the old method at all.
(this was an accidental compatibility breakage, actually; method should have been marked as Deprecated, and only removed in 3.0... but was not reported by anyone prior to release of 2.10 AFAIK).

So leaving use of the old method is unfortunately not an option.

Reply:
With 2.12.5 I can revert this change as I added back the method in question for backwards-compatibility.
Ideally (in future) call will be changed, but it is lower risk doing minimal change.